apps/evm/src/libs/contracts/utilities/getContractAddress/index.ts

```diff
@@ -13,14 +13,14 @@ export type GetUniqueContractAddressInput = {
 export type GetUniquePerPoolContractAddressInput = {
   name: UniquePerPoolContractName;
   chainId: ChainId;
-  poolComptrollerContractAddress: string;
+  poolComptrollerContractAddress: Address;
 };
 
 export type GetContractAddressInput =
   | GetUniqueContractAddressInput
   | GetUniquePerPoolContractAddressInput;
 
-export const getContractAddress = (input: GetContractAddressInput) => {
+export const getContractAddress = (input: GetContractAddressInput): Address | undefined => {
```

Comment:
Doing this is riskier because you cast the return type of the whole function. Ideally we won't to cast as less as possible of the type, hence my original implementation.

Reply:
Removed this and the `keyof typeof` casts in favor of generating strict types. The only exception now is the `toLowerCase()` call, which makes the key get typed to `string` instead of `Address`.